server/src/main/java/org/elasticsearch/common/logging/DeprecatedMessage.java

```diff
@@ -31,13 +33,14 @@
     public static final String ECS_VERSION = "1.6";
 
     @SuppressLoggerChecks(reason = "safely delegates to logger")
-    public static ESLogMessage of(String key, String xOpaqueId, String messagePattern, Object... args) {
+    public static ESLogMessage of(DeprecationCategory category, String key, String xOpaqueId, String messagePattern, Object... args) {
         ESLogMessage esLogMessage = new ESLogMessage(messagePattern, args)
             .field("data_stream.type", "logs")
             .field("data_stream.dataset", "deprecation.elasticsearch")
             .field("data_stream.namespace", "default")
             .field("ecs.version", ECS_VERSION)
-            .field("key", key);
+            .field("key", key)
+            .field("elasticsearch.event.category", category.name().toLowerCase(Locale.ROOT));
```

Comment:
Why not just `category`? Is this ECS related?

Reply:
Yes, the broader change is in #67266.